replication-api/src/main/java/org/codice/ditto/replication/api/ReplicatorConfig.java

```diff
@@ -46,27 +56,78 @@
    */
   ReplicationType getReplicationType();
 
+  /**
+   * Set the type of replication to be performed metacard or resource. Resource replication includes
+   * metacard replication.
+   *
+   * @param replicationType the new replication type
+   */
+  void setReplicationType(ReplicationType replicationType);
+
   /**
    * Get the source site id associated with this configuration
    *
    * @return source id
    */
   String getSource();
 
+  /**
+   * Set the source site id associated with this configuration
+   *
+   * @param sourceId the id for the source site
+   */
+  void setSource(String sourceId);
+
   /**
    * Get the destination site id associated with this configuration
    *
    * @return destination id
    */
   String getDestination();
 
+  /**
+   * Set the destination site id associated with this configuration
+   *
+   * @param destinationId the id for the destination site
+   */
+  void setDestination(String destinationId);
+
   /**
    * Get the cql filter used for determining the data set to replicate
    *
    * @return replication cql filter
    */
+  @Deprecated
```

Comment:
This was a case of me trying to avoid making changes in a lot of places. I wanted to refactor it to be called getFilter but didn't want to mess with all the files it was being used in. So, I just added a getFilter method and deprecated the old one. Looking back on all the other refactoring I did, I think I'll just go back and rename it.